packages/fxa-settings/src/pages/PostVerify/ThirdPartyAuthCallback/index.tsx

```diff
@@ -72,12 +75,39 @@ const ThirdPartyAuthCallback = ({
   /**
    * Navigate to the next page
    * if Sync based integration -> navigate to set password or sign-in
-   * if OAuth based integration -> verify OAuth and navigate to RP
+   * If browser login without keys -> navigate to settings
+   * if OAuth web based integration -> verify OAuth and navigate to RP
    * if web redirect (SubPlat) -> navigate back to SubPlat
    * if neither -> navigate to settings
    */
   const performNavigation = useCallback(
     async (linkedAccount: LinkedAccountData, needsVerification = false) => {
+      const shouldSendWebChannelMessages =
+        integration.isFirefoxClientServiceRelay() ||
+        integration.isFirefoxClientServiceAiMode();
+
+      if (shouldSendWebChannelMessages) {
+        const { ok } = await firefox.fxaCanLinkAccount({
+          email: linkedAccount.email,
+        });
+        if (!ok) {
+          // User cancelled the login, redirect back to Index with error banner.
+          // Prefill the email with this account to prevent a redirect to '/signin'
+          // so they can see the error and decide what to do.
+          navigateWithQuery('/', {
+            replace: true,
+            state: {
+              localizedErrorFromLocationState: ftlMsgResolver.getMsg(
+                'auth-error-1001',
+                'Login cancelled'
```

Comment:
At this stage, aren't they signed in on web but not in the browser?

Reply:
Yes. It's a little awkward, but they did cancel the _Sync_ login attempt, and this is the earliest place we will have the account email / created UID (will be needed for secondary email signins) and should be an edge case.